js-core/homeworks/homework-4/src/main.js

```diff
@@ -0,0 +1,124 @@
+/*
+* Напишите функцию которая будет принимать 1 аргумент и в зависимости от типа аргумента
+* Если тип аргумента число или объект -> возвращать true
+* Если тип аргумента функция -> возвращать false
+* Если тип аргумента Строка и длина этой строки не равна 10 -> возвращать "длина вашей строки: <длина строки>
+* Если длина 10 -> 'you win'
+* */
+let obj = {
+    name: 'aloe'
+};
+let num = 15;
+let stringALotLength = 'aloe vera lorem ipsum';
+let stringWithLength10 = 'aloe vera.';
+let func = function(){
+    return 'yes it is';
+};
+
+function check(param){
+    let isNumOrObject = typeof param === 'number' || typeof param === 'object';
+    let isFunction = typeof param === 'function';
+    let isString = typeof param === 'string';
+    let stringLength10 = param.length === 10;
+    let stringLengthNot10 = param.length != 10;
+    if(isNumOrObject){
+        return true;
+    }
+    else if(isFunction){
+        return false;
+    }
+    else if(isString && stringLengthNot10){
+        return 'Длина вашей строки:' + param.length;
+    }
+    else if(isString && stringLength10){
+        return 'you win';
+    }
+    return param;
+};
+
+console.log(check(obj));
+console.log(check(num));
+console.log(check(stringALotLength));
+console.log(check(stringWithLength10));
+console.log(check(func));
+//==============================================================================================
+/*
+* Напишите функцию которая принимает 2 числа
+* и возвращает массив содержащий числа между первым числом и вторым числом;
+ */
+ function betweenNumbers(number1,number2){
+    let array = [];
+    for(let i = number1; i < number2 +1; i++){
+        array[i - number1] = i;
+    }    
+    return array;
+ };
+ console.log(betweenNumbers(25,64));
+//==============================================================================================
+/*
+ 2. Перепишите задачу FizzBuzz, но с использованием цикла.
+ Расчет чисел должен идти до 100
+ */
+function fizzBuzz100(num){    
+    let str;
+    for(let i = 0; i < 101; i++){        
+        if(i % 3 === 0 && i % 5 === 0){
```

Comment:
I think we have to move to another function finding if num returns Fizz, Buzz or FizzBuzz.
Or you can re-use existing solution from previous homework